### ISSUE TITLE
Any possibility to add in Pro features?

### ISSUE THREAD

**Issue by lingnand** [hzqtc/fmd#11]
Just let you know that your project looks really interesting. Is there any plan to implement Pro features such as high-quality music and downloads?

**Comment by lingnand** [comment]
I've tried to experiment a bit and come up with this patch. Now we can set the kbps info in the .conf at https://gist.github.com/yulan6248/5630013

**Comment by hzqtc** [comment]
Cool! I examed the your patch. The code is clean.

But I can't verify it because I'm not a pro user and I get warning message in API response with `kbps` parameter. The message is (no matter `kbps=64` or `kbps=192`):

> "warning": "user_is_ananymous"

Well, at least I know `kbps` parameter does exist. You must be a pro user. Have you verify this?

I don't want to including downloading in FMD. FMD is supposed to remain small and simple.

I would like to merge your patch. Would you kindly use the **fork & pull request** feature of github which will make merge much easier. Also remember to edit `README.md` and `fmd-update-conf.sh` correspondingly.

Thanks!

**Comment by lingnand** [comment]
I'm a Pro user and for me it's working fine :) I read somewhere that if it's not a Pro user then this parameter would not do anything. So I guess the best way is to verify if the user is a Pro user and then add the parameter only when necessary, but I'm too lazy for that. 

I'm still quite new to Github but it looks interesting and I'd do fork & pull in a minute

**Comment by hzqtc** [comment]
The login API don't return any information about whether a user is pro. So we just set the default `kbps` to ` `(empty) and notice pro users to change it to`192`, before we can figure out a decent way to detect pro users.

Cheers!

**Comment by lingnand** [comment]
Oh sorry I missed your message and I've already made the pull request...When you say set it to empty do you mean passing the "kbps=" in the url request?

**Comment by lingnand** [comment]
I'm a little unsure about what you said. From what I see it seems the kbps field will get some random value in the end if we just leave it blank. Probably setting it to zero and having a condition in 'send report' to check for that?

**Comment by hzqtc** [comment]
That's OK.
I mean the config value of `kbps` should be a string so that we can left it empty in the config file. Because for non-pro users, we got a warning message whenever value we specify as `kbps` (e.g. 0, 64 or 192).

So I will merge your pull request and edit a little bit.

**Comment by hzqtc** [comment]
If its type is int, yes, it will get random value if left empty in config file. So I decided to make it a string.

I prefer to make all config values strings unless that are explicitly refered as int in the code. If we only pass a config value in the URL, why bother to convert it to int and then `sprintf` it to string later?

That the same reason why I set `port` config type to string: `getaddrinfo` accept `char *` port numbers.

Actually, the user does not need to know config types any way. Maybe I should not describe config value type in README. :)
